security/pkg/server/ca/server.go

```diff
@@ -68,13 +82,34 @@ func (s *Server) CreateCertificate(ctx context.Context, request *pb.IstioCertifi
 		s.monitoring.AuthnError.Increment()
 		return nil, status.Error(codes.Unauthenticated, "request authenticate failure")
 	}
-	// TODO: Call authorizer.
+	// By default, we will use the callers identity for the certificate
+	sans := caller.Identities
```

Comment:
Is the caller ztunnel? What identity is it using?

Reply:
Yes, it uses its own identity

Reply:
Do we need to verify here in the code that this is ztunnel identity?

Reply:
We do in the node authorizer, that is what trustedNodeAccounts is

Reply:
Other than verifying that the service account is kube-system/ztunnel, do we also check the node id, etc? I guess my question is, how do we prevent ztunnel on one node to impersonate a workload on another node?

Reply:
Did you read the code? Every line in the authorizer is thoroughly commented explaining this.

Checking the node is the entire point of this PR